test/unit/math/prim/mat/fun/matrix_exp_compute_test.cpp

```diff
@@ -0,0 +1,39 @@
+#include <stan/math/prim/mat.hpp>
+#include <stan/math/prim/mat/fun/MatrixExponential.h>
+#include <gtest/gtest.h>
+#include <test/unit/math/prim/mat/fun/expect_matrix_eq.hpp>
+
+using Eigen::Matrix;
+using Eigen::Dynamic;
+using stan::math::matrix_exp_compute;
+
+TEST(MathMatrix, matrix_exp_compute_1) {
+    
+    Matrix<double, Dynamic, Dynamic> m1(1,1), m2(1,1);
+    m1 << 0;
+    m2 << 1;
+        
+    expect_matrix_eq(m2, matrix_exp_compute(m1));
+}
+
+TEST(MathMatrix, matrix_exp_compute_2) {
+    
+    // example from http://www.sosmath.com/matrix/expo/expo.html
+    Matrix<double, Dynamic, Dynamic> m1(3,3), m2(3,3), m3(3,3), I(3,3);
+    m1 << 0, 1, 2, 0, 0, -1, 0, 0, 0;
+    m2 << 1, 1, 1.5, 0, 1, -1, 0, 0, 1;
+    
```

Comment:
Don't define variables like `m3` or `I` that you don't use.  And push the declarations as far down as you can in the code before you use variables---it makes the code much easier to see becuase it's easier to find declarations.